### ISSUE TITLE
Can't pair Trezor T device

### ISSUE THREAD

**Issue by mrpelotazo** [input-output-hk/daedalus#2553]
The **Pair a hardware wallet device** dialog is stuck with the **Connect your device and enter your PIN to unlock it** message even after unlocking the Trezor device.

Currently running on
Daedalus 4.0.5 
trezord 2.0.30
Arch Linux (5.11.16-arch1-1)

I have no issues with other application accessing the device over the trezor bridge.
Deleting Daedalus state directory and reinstalling the application doesn't help.

On the terminal I get following log output

```
{"at":"2021-05-06T15:05:07.643Z","env":"mainnet:Linux:5.11.16-arch1-1","ns":["daedalus","v4.0.5","*mainnet*"],"data":{},"app":["daedalus"],"msg":"[HW-DEBUG] getHardwareWalletTransportChannel","pid":"","sev":"info","thread":""}
{"at":"2021-05-06T15:05:07.643Z","env":"mainnet:Linux:5.11.16-arch1-1","ns":["daedalus","v4.0.5","*mainnet*"],"data":{},"app":["daedalus"],"msg":"[HW-DEBUG] getHardwareWalletTransportChannel::TREZOR ","pid":"","sev":"info","thread":""}
{"at":"2021-05-06T15:05:07.643Z","env":"mainnet:Linux:5.11.16-arch1-1","ns":["daedalus","v4.0.5","*mainnet*"],"data":{},"app":["daedalus"],"msg":"[HW-DEBUG] Trezor connect success","pid":"","sev":"info","thread":""}
{"at":"2021-05-06T15:05:07.643Z","env":"mainnet:Linux:5.11.16-arch1-1","ns":["daedalus","v4.0.5","*mainnet*"],"data":{},"app":["daedalus"],"msg":"[HW-DEBUG] Trezor connect error","pid":"","sev":"info","thread":""}
node-usb [../src/node_usb.cc:handleHotplug() 130]: HandleHotplug 0x3a73ec710960 1
node-usb [../src/device.cc:Device() 17]: Created device 0x3a73ecb44990
node-usb [../src/node_usb.cc:handleHotplug() 138]: Device arrived
node-usb [../src/device.cc:~Device() 21]: Freed device 0x3a73ecb44990
```

trezord log output

```
May 06 17:23:57 ******** trezord[640]: POST /
May 06 17:23:57 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:23:57 +0200] "POST / HTTP/1.1" 200 21
May 06 17:23:58 ******** trezord[640]: POST /enumerate
May 06 17:23:58 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:23:58 +0200] "POST /enumerate HTTP/1.1" 200 3
May 06 17:23:58 ******** trezord[640]: POST /listen
May 06 17:25:00 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:23:58 +0200] "POST /listen HTTP/1.1" 200 94
May 06 17:25:00 ******** trezord[640]: POST /listen
May 06 17:25:00 ******** trezord[640]: POST /acquire/7/null
May 06 17:25:01 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:25:00 +0200] "POST /acquire/7/null HTTP/1.1" 200 16
May 06 17:25:01 ******** trezord[640]: POST /call/8
May 06 17:25:01 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:25:01 +0200] "POST /call/8 HTTP/1.1" 200 372
May 06 17:25:01 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:25:00 +0200] "POST /listen HTTP/1.1" 200 93
May 06 17:25:01 ******** trezord[640]: POST /listen
May 06 17:25:01 ******** trezord[640]: POST /release/8
May 06 17:25:01 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:25:01 +0200] "POST /listen HTTP/1.1" 200 94
May 06 17:25:01 ******** trezord[640]: POST /listen
May 06 17:25:01 ******** trezord[640]: 127.0.0.1 - - [06/May/2021:17:25:01 +0200] "POST /release/8 HTTP/1.1" 200 16
```

**Comment by nikolaglumac** [comment]
I believe this is due to your local system configuration. Trezor works normally on Ubuntu (which we officially support: https://iohk.zendesk.com/hc/en-us/articles/360010496553-Daedalus-System-Requirements).
Perhaps it is bindings issue or something specific to Arch Linux, but I am not sure.
Please make sure you do not have any active connections to the device before trying to pair it with Daedalus - e.g. restart your machine and then try to pair it with Daedalus as your first action.

**Comment by nikolaglumac** [comment]
@ManusMcCole are you able to spin up a Arch Linux instance and test pairing Trezor model T there? 🙏

**Comment by mrpelotazo** [comment]
> I believe this is due to your local system configuration.

Sure, its possible, but on the other hand other applications are working (trezor-suite, metamask), so (at this stage) I'm inclined to think that the differentiating factor relies somewhere else.

> Trezor works normally on Ubuntu (which we officially support: https://iohk.zendesk.com/hc/en-us/articles/360010496553-Daedalus-System-Requirements).

From what I read there, daedalus officially is "Aimed at all Linux distributions".

> Perhaps it is bindings issue or something specific to Arch Linux, but I am not sure.

Neither am I. I'm sure we''ll figure it out. Maybe the bundled trezor-connect version has a play in this(?)

> Please make sure you do not have any active connections to the device before trying to pair it with Daedalus - e.g. restart your machine and then try to pair it with Daedalus as your first action.

Done, no success.

@ManusMcCole  I've run the trezord daemon in verbose mode (logs attached) with both the trezor suite and daedalus. In both cases no prior connection to the trezor device was made. I thought it might be helpful for you guys to have a more detailed log.

[trezord-daedalus.log](https://github.com/input-output-hk/daedalus/files/6442664/trezord-daedalus.log)
[trezord-trezorsuite.log](https://github.com/input-output-hk/daedalus/files/6442665/trezord-trezorsuite.log)

**Comment by nikolaglumac** [comment]
Keep in mind Daedalus is sandboxed and perhaps that is the issue here. We need support from our DEVOPS crew to investigate this. @disassembler @cleverca22 please help us out 🙏

**Comment by nikolaglumac** [comment]
@mrpelotazo can you please try pairing your Trezor device with the latest Daedalus Mainnet 4.2.0 version?

**Comment by mrpelotazo** [comment]
@nikolaglumac It seems Daedalus was affected by the environment set by NVM either during the installation or when run after.
I ended up completely disabling NVM in my bashrc, made a fresh Daedalus installation and the issue was solved.

I don't know if there is an effective way for you guys to detect/manage such situations somehow. To my knowledge NVM is a relatively popular thing to use among developers using nodejs, so, from a user perspective, it would be very helpful to maybe at least get a warning of some kind if NVM is detected in the environment. Ideally Daedalus would't get affected by the presence of NVM at all.
